### PR TITLE
Introduce caxis to concepts and functions

### PR DESCRIPTION
**Pull request by f-brinkmann**
### Which issue(s) are closed by this pull request?

As discussed in the weekly

### Changes proposed in this pull request:

- add `cshape` and `caxis` to audio class concepts
- add caxis to `dsp.average` and `matrix_multiplication`

### CODE REVIEW

**Review by sikersten — APPROVED** [review]

File: pyfar/dsp/dsp.py [hunk 1]
```diff
@@ -1880,12 +1880,11 @@ def average(signal, mode='linear', axis=None, weights=None, keepdims=False):
...
 
         The default is ``'linear'``
-    axis: None, int, or tuple of ints, optional
-        Axis or axes along which the averaging is done. Can be ``None``, which
-        will calculate the average across all channel axes. Negative values
-        refer to ``signal.cshape`` to avoid averaging across the time or
-        frequency axis. If axis is a tuple of ints, average will perform on the
-        channels specified in the tuple. The default is ``None``.
+    caxis: None, int, or tuple of ints, optional
+        Channel axis or axes along which the averaging is done. The default
```
Comment (mberz) [inline pyfar/dsp/dsp.py:1884]:
```suggestion
        Channel axes along which the averaging is done. The default
```
Pl. alone should suffice here

File: pyfar/dsp/dsp.py [hunk 2]
```diff
@@ -1880,12 +1880,11 @@ def average(signal, mode='linear', axis=None, weights=None, keepdims=False):
...
+        ``None``, average across all channel axes. See
```
Comment (mberz) [inline pyfar/dsp/dsp.py:1885]:
```suggestion
        ``None``, average across all channels. See
```

File: pyfar/dsp/dsp.py [hunk 3]
```diff
@@ -1921,21 +1920,23 @@ def average(signal, mode='linear', axis=None, weights=None, keepdims=False):
...
-    # check for axis
-    if axis and np.max(axis) > len(signal.cshape):
-        raise ValueError('The maximum of axis needs to be smaller then '
-                         'len(signal.cshape).')
-    # set axis default
-    if axis is None:
-        axis = tuple([i for i in range(len((signal.cshape)))])
+    # check for caxis
+    if caxis and np.max(caxis) > len(signal.cshape):
+        raise ValueError(('The maximum of caxis needs to be smaller then '
```
Comment (mberz) [inline pyfar/dsp/dsp.py:1925]:
```suggestion
        raise ValueError(('The maximum of caxis needs to be smaller than '
```

**Review by mberz — CHANGES_REQUESTED** [review]
Wording and language.
You're trying to explain the caxis using axis, maybe matrix terminology is better suited and easier to understand from a user perspective.

**Pull request merged**